bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs

```diff
@@ -97,29 +83,34 @@ public async Task<IEnumerable<Core.SecretsManager.Entities.Secret>> GetManyByOrg
                                     .OrderBy(c => c.RevisionDate)
                                     .ToListAsync();
 
-            return Mapper.Map<List<Core.SecretsManager.Entities.Secret>>(secrets);
+            // This should be changed if/when we allow non admins to access trashed items
+            return Mapper.Map<List<Core.SecretsManager.Entities.Secret>>(secrets).Select(s => new SecretPermissionDetails
+            {
+                Secret = s,
+                Read = true,
+                Write = true,
+            });
         }
     }
 
-    public async Task<IEnumerable<Core.SecretsManager.Entities.Secret>> GetManyByProjectIdAsync(Guid projectId, Guid userId, AccessClientType accessType)
+    public async Task<IEnumerable<SecretPermissionDetails>> GetManyByProjectIdAsync(Guid projectId, Guid userId, AccessClientType accessType)
     {
-        using (var scope = ServiceScopeFactory.CreateScope())
+        using var scope = ServiceScopeFactory.CreateScope();
+        var dbContext = GetDatabaseContext(scope);
+        var query = dbContext.Secret.Include(s => s.Projects)
+            .Where(s => s.Projects.Any(p => p.Id == projectId) && s.DeletedDate == null);
+
+        query = accessType switch
         {
-            var dbContext = GetDatabaseContext(scope);
-            var query = dbContext.Secret.Include(s => s.Projects)
-                .Where(s => s.Projects.Any(p => p.Id == projectId) && s.DeletedDate == null);
+            AccessClientType.NoAccessCheck => query,
+            AccessClientType.User => query.Where(UserHasReadAccessToSecret(userId)),
+            AccessClientType.ServiceAccount => query.Where(ServiceAccountHasReadAccessToSecret(userId)),
+            _ => throw new ArgumentOutOfRangeException(nameof(accessType), accessType, null),
+        };
 
-            query = accessType switch
-            {
-                AccessClientType.NoAccessCheck => query,
-                AccessClientType.User => query.Where(UserHasReadAccessToSecret(userId)),
-                AccessClientType.ServiceAccount => query.Where(ServiceAccountHasReadAccessToSecret(userId)),
-                _ => throw new ArgumentOutOfRangeException(nameof(accessType), accessType, null),
-            };
+        var secrets = SecretToPermissionDetails(query, userId, accessType);
```

Comment:
It looks like `SecretToPermissionDetails` already handles adding the `UserHasReadAccessToSecret` and `ServiceAccountHasReadAccessToSecret` where clauses for us and lines 105-109 aren't needed?

But maybe I'm missing something.

Reply:
Are you thinking we should do `SecretsToPermissionDetails` first, and then just check `secrets.Where(s => s.Read)`?

Reply:
In `SecretsToPermissionDetails`

```
    private IQueryable<SecretPermissionDetails> SecretToPermissionDetails(IQueryable<Secret> query, Guid userId, AccessClientType accessType)
    {
        var secrets = accessType switch
        {
            AccessClientType.NoAccessCheck => query.Select(s => new SecretPermissionDetails
            {
                Secret = Mapper.Map<Bit.Core.SecretsManager.Entities.Secret>(s),
                Read = true,
                Write = true,
            }),
            AccessClientType.User => query.Where(UserHasReadAccessToSecret(userId)).Select(SecretToPermissionsUser(userId, true)),
            AccessClientType.ServiceAccount => query.Where(ServiceAccountHasReadAccessToSecret(userId)).Select(s =>
                new SecretPermissionDetails
                {
                    Secret = Mapper.Map<Bit.Core.SecretsManager.Entities.Secret>(s),
                    Read = true,
                    Write = false,
                }),
            _ => throw new ArgumentOutOfRangeException(nameof(accessType), accessType, null),
        };
        return secrets;
    }
```

For the user, a where query for the `UserHasReadAccessToSecret` and the Select
```
AccessClientType.User => query.Where(UserHasReadAccessToSecret(userId)).Select(SecretToPermissionsUser(userId, true))
```


For the service account, a where query for the `ServiceAccountHasReadAccessToSecret` and the Select
```
AccessClientType.ServiceAccount => query.Where(ServiceAccountHasReadAccessToSecret(userId)).
```


Looks to be doing the same thing as lines 103-109 in `GetManyByProjectIdAsync`

```
  query = accessType switch
        {
            AccessClientType.NoAccessCheck => query,
            AccessClientType.User => query.Where(UserHasReadAccessToSecret(userId)),
            AccessClientType.ServiceAccount => query.Where(ServiceAccountHasReadAccessToSecret(userId)),
            _ => throw new ArgumentOutOfRangeException(nameof(accessType), accessType, null),
        };
```

So all combined, we're doing the where query twice? 

```
AccessClientType.User => query.Where(UserHasReadAccessToSecret(userId)).Where(UserHasReadAccessToSecret(userId)).Select(SecretToPermissionsUser(userId, true))
```

Reply:
Correct, I've removed the duplicated condition.